common/date_utils.py

```diff
@@ -315,3 +317,39 @@ def merge_time_slots(time_slots: list[TimeSlot]) -> list[TimeSlot]:
         # Otherwise the periods are not contiguous -> append the period and continue.
         merged_slots.append(period)
     return merged_slots
+
+
+def get_periods_between(
+    start_date: datetime.date,
+    end_date: datetime.date,
+    start_time: datetime.time,
+    end_time: datetime.time,
+    *,
+    interval: int = 7,
+    tzinfo: zoneinfo.ZoneInfo | datetime.timezone | None = None,
+) -> Generator[tuple[datetime.datetime, datetime.datetime], None, None]:
+    """
+    Generate datetimes based on the given start and end dates.
+
+    Note: This will generate the times in such a way that the time will be the same after
+          switching to daylight saving time and back.
+
+    :param start_date: From which date to generate dates?
+    :param end_date: Until which date to generate dates?
+    :param start_time: Start time of a period.
+    :param end_time: End time of a period.
+    :param interval: Days between each period.
+    :param tzinfo: Timezone information for the datetimes (if times are not datetime-aware yet).
+    """
+    if end_date < start_date:
+        msg = "End date cannot be before start date."
+        raise ValueError(msg)
+    if end_time < start_time:
```

Comment:
Start and end time cannot be equal either.

Reply:
Fixed, and added tests for the utility